### PR TITLE
161 restore api calls

### PR DESCRIPTION
**Pull request by alishaevn**
# story
as a signed in user, no api calls were returning data. with this pr, they will

# expected behavior
- the github ui would not let me revert #159, so I am doing it manually. now that we always have an access token that belongs to the provider or the user, we do not need to check for the existence of one. which means we do not get stuck in loading because the api is using a "null" value
- only conditionally check for api params that are not the access token
- return the loading component first so the page does not glitch for a signed in user

# demo

https://user-images.githubusercontent.com/29032869/216103654-135396ef-df64-46e5-b68c-a84d8b3795f6.mp4

### CODE REVIEW

File: pages/requests/[id].js [hunk 1]
```diff
@@ -34,6 +34,8 @@ const Request = () => {
   const isLoading = isLoadingRequest || isLoadingSOWs || isLoadingMessages
   const isError = isRequestError || isSOWError || isMessagesError
 
+  if (isLoading) return <Loading wrapperClass='item-page mt-5' />
```
Comment (alishaevn) [inline pages/requests/[id].js:37]: moved our loading state to come before checking for a user so that the page doesn't glitch for a signed in user.

File: pages/requests/new/[ware].js [hunk 2]
```diff
@@ -111,7 +111,9 @@ const NewRequest = () => {
   // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [requestSucceeded, requestErred, requestID])
 
-  // TODO(alishaevn): update the return value after https://github.com/scientist-softserv/webstore-component-library/issues/136 is completed
+  // TODO(alishaevn): use react bs placeholder component
+  if (isLoadingInitialRequest || !wareID) return <Loading wrapperClass='item- mt-5' />
```
Comment (summer-cook) [inline pages/requests/new/[ware].js:115]: this class name of `item-` doesn't seem right, just double checking with you that was intentional?

**Review by summer-cook — COMMENTED** [review]
I just checked this branch and this work brings back 403 console errors on every page

image

image

image

image


originally the [conditional fetching](https://swr.vercel.app/docs/conditional-fetching) was added so that wouldn't happen. is there a way to possibly refactor the conditional fetching, or otherwise prevent these errors from happening?

**Comment by summer-cook** [comment]
> I just checked this branch and this work brings back 403 console errors on every page
> 
> image
> 
> image
> 
> image
> 
> image
> 
> originally the [conditional fetching](https://swr.vercel.app/docs/conditional-fetching) was added so that wouldn't happen. is there a way to possibly refactor the conditional fetching, or otherwise prevent these errors from happening?

also looking more at it, maybe its just that the pages that can only be seen by a logged IN user do still need accessToken to be checked?

I am also a bit confused because i thought with the way the auth works, logged out users will have a client credential... so maybe those calls need to check for the client credential vs the access token?

**Comment by alishaevn** [comment]
> I just checked this branch and this work brings back 403 console errors on every page
> ...
> originally the [conditional fetching](https://swr.vercel.app/docs/conditional-fetching) was added so that wouldn't happen. is there a way to possibly refactor the conditional fetching, or otherwise prevent these errors from happening?

tldr -- no. conditional fetching won't work the way as described when the condition is an access token. also, the errors (that I'm seeing anyway) are only in relation to getting a promo image, and they're inconsistent at best.

detailed answer -- 
your chrome console seems to only show you the failures, but not the successes. in firefox dev edition, I'm getting both. so I can see that an api call originally failed, then succeeded. as seen below. idk why only sometimes promo image fail initially but then succeed. and why it only happens to a random image and not all of them. 🤷🏾‍♀️ we can hop on a call to look at more examples since that'll be easier.

the conditional fetching you used is the same as seen in this [old code](https://github.com/scientist-softserv/webstore/blob/95961d9bfe0669b95d5219819cec3f49a9fbd7e8/pages/index.js#L17-L24)). however, it requires that the condition being checked against become true at some point. prior to us setting up auth we wouldn't have had this problem because we always had an access token in our env variables. in the case of a logged out user, there will never be a session and therefore there won't be an access token to pass to the api. so the conditional will always be false, which means `null` will be the outcome and `useSWR` will never be invoked. hence, our perpetual loading state.

the issue with the promo image is that `ware.urls.promo_image` (being used in `configureServices`) is a url. so as we try to access it, sometimes it seems to not respond correctly. there isn't a request, response or status on the call when it fails so idk why it happens. but it only happens sometimes. for me. one refresh will have no errors, the next will.

it doesn't appear to actually be messing up the app though.

image

File: pages/requests/new/[ware].js (hunk 2 above)
Comment (alishaevn) [reply pages/requests/new/[ware].js:115]: good catch! I had to go back a few commits to figure out what this is actually supposed to be. it was removed in [this commit](https://github.com/scientist-softserv/webstore/commit/fd56f5d085d4a3671349278f7d7ed82f8d526a7a) last week.

Comment (alishaevn) [reply pages/requests/new/[ware].js]: fixed!

**Review by summer-cook — APPROVED** [review]

**Comment by alishaevn** [comment]
the discrepancy wasn't chrome vs. firefox dev. it was because summer didn't have the client access token in local dev. once I gave it to her, her errors went away.
